src/Raven.Studio/typescript/viewmodels/database/indexes/editIndex.ts

```diff
@@ -247,6 +264,88 @@ class editIndex extends viewModelBase {
             });
     }
 
+    private fetchIndexHistory() {
+        const db = this.activeDatabase();
+        
+        return new getIndexHistoryCommand(db, this.editedIndex().name() || this.originalIndexName)
+            .execute()
+            .done((indexHistory) => {
+                const indexHistoryInfo = indexHistory.History.map(x => {
+                    return { CreatedAt: x.CreatedAt, Definition: x.Definition, Source: x.Source, CreatedAtFormatted: generalUtils.formatUtcDateAsLocal(x.CreatedAt) };
+                });
+                
+                this.indexHistory(indexHistoryInfo);
+            });
+    }
+    
+    static formatTime(input: string) {
+        const dateToFormat = moment.utc(input);
+        return dateToFormat.format(editIndex.indexHistoryCreatedAtFormat) + "Z";
+    }
+
+    useIndexRevisionItem(item: indexHistoryDefinition) {
+        this.previewItem(item);
+        this.loadIndexDefinitionFromHistory();
+    }
+    
+    loadIndexDefinitionFromHistory() {
+        const newIndexDefinition = new indexDefinition(this.previewItem().Definition);
+
+        if (!this.isEditingExistingIndex()) {
+            // if editing a clone then load the index definition without the index name
+            newIndexDefinition.name(null);
+        }
+        
+        this.editedIndex(newIndexDefinition);
+        
+        this.loadedIndexHistory(true);
+    }
+
+    loadMapReduceFromHistory() {
+        const previewItem = this.previewItem();
+        const mapsFromPreview = previewItem.Definition.Maps;
+        const reduceFromPreview = previewItem.Definition.Reduce;
+        
+        const newIndexDefinition = new indexDefinition(this.editedIndex().toDto());
+        newIndexDefinition.setMapsAndReduce(mapsFromPreview, reduceFromPreview);
+        this.editedIndex(newIndexDefinition);
+
+        this.loadedIndexHistory(true);
+    }
+
+    indexHistoryButtonHandler() {
+        this.fetchIndexHistory().done(() => {
+            this.previewItem(this.indexHistory()[0]);
+
+            $('.history-list [data-toggle="tooltip"]').tooltip({
+                html: true
+            });
+        });
+    }
+
+    previewIndex(item: indexHistoryDefinition) {
+        this.previewItem(item);
+    }
+
+    creationTimeTooltip(item: indexHistoryDefinition) {
+        return ko.pureComputed(() => {
+            if (item) {
+                return `<div class="date-container">
+                            <div>
+                                <div class="date-label">UTC:</div>
+                                <div class="date-value">${moment.utc(item.CreatedAt).format(editIndex.indexHistoryCreatedAtFormat) + "Z"}</div>
```

Comment:
and also here you won't need to format that.

Reply:
done